fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -481,9 +485,34 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
         }
     }
 
+    private fun subscribeToAccountManager() {
+        requireComponents.backgroundServices.accountManagerAvailableQueue.runIfReadyOrQueue {
+            // By the time this code runs, we may not be attached to a context or have a view lifecycle owner.
+            if ((this@BrowserFragment).view?.context == null) {
```

Comment:
I am not sure I understand the question, but since we are waiting for a queue, the fragment might be closed by the time it finishes. Is the approach wrong?

Reply:
In fact, if we use the `SyncStore`, then you don't even need the `runIfReadyOrQueue` because you would be observing the store and not eagerly initializing the `AccountManager`, so this way, you will always have the `Context` available (because that's when you start observing, and you will be able to see the state when the account manager is eventually initialized.

```kotlin
class AccountDeletionSnackbarObserver(
    private val containerView: View,
    private val anchorView: View,
    private val store: SyncStore,
) : AbstractBinding<SyncState>(store) {
    override suspend fun onState(flow: Flow<SyncState>) {
        flow.map { it.status }
            .distinctUntilChanged()
            .collectLatest { status ->
                if (status == SyncStatus.Deleted) {
                    FenixSnackbar.make(
                        view = containerView,
                        duration = FenixSnackbar.LENGTH_SHORT,
                        isDisplayedWithBrowserToolbar = true,
                    )
                        .setText(containerView.context.getString(R.string.snackbar_account_deleted))
                        .setAnchorView(anchorView)
                        .show()

                    // Consume the state so we don't keep showing the snackbar.
                    // Maybe this should be done somewhere else since changing the sync status should not be specific
                    // to the snackbar observer.
                    store.dispatch(SyncAction.UpdateSyncStatus(SyncStatus.NotInitialized))
                }
            }
    }
}
```

Reply:
If we stick to this approach, the resetting of `SyncStatus` should happen after after the `Snackbar` is displayed, which doesn't sound right to me, because change of the state has nothing to do with the snackbar display. Adding a new `SyncStatus` might make sense for different cases, but in this particular one we still have a problem of knowing if we should show a snackbar or not when `status == SyncStatus.Deleted`

What I think is happening, we will be introducing a `SyncStatus.Deleted` status as a channel to pass down the account deleted event, which isn't the cleanest approach either. The cleaner approach would be to intoduce a `SnackbarManager`, that would queue the events and pass it down to subscribers to handle, the system that would receive an even from `AccountObserver.onAccountDeleted`, but wouldn't meddle with the `SyncStore` state. But since we are not going for it, why not just stick to a global flag for this case? Setting it to true and resetting after snackbar display seems like a cleaner approach than utilizing a new `SyncState` for that.

Reply:
> If we stick to this approach, the resetting of SyncStatus should happen after after the Snackbar is displayed, which doesn't sound right to me, because change of the state has nothing to do with the snackbar display.

Yes, this is the problem that I mentioned in my comment in the code snippet. We can still solve this though.

> What I think is happening, we will be introducing a SyncStatus.Deleted status as a channel to pass down the account deleted event, which isn't the cleanest approach either. The cleaner approach would be to intoduce a SnackbarManager, that would queue the events and pass it down to subscribers to handle, the system that would receive an even from AccountObserver.onAccountDeleted, but wouldn't meddle with the SyncStore state.

We need to update the SyncStore with the new deleted state regardless, because we want the Store to be the new way to observe the store and not use the AccountManager directly. So since we have to do that, we could then use it to for the snackbar case as well.

> But since we are not going for it, why not just stick to a global flag for this case? Setting it to true and resetting after snackbar display seems like a cleaner approach than utilizing a new SyncState for that.

We have established patterns of how we hold and observe state that we know works, so if we're going to create a new pattern, we should have strong reasons for it.

If you prefer using a global flag, then please put it in the AppStore (established pattern) and observe that. As mentioned above, we do still need to keep the SyncStore up-to-date.